docs/pages/deepsleep-manager_TestSpec.md

```diff
@@ -0,0 +1,83 @@
+# Deepsleep Manager Test Specification Documentation
+
+## Table of Contents
+
+- [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+- [Introduction](#introduction)
+- [Module Description](#module-description)
+- [Testing Scope](#testing-scope)
+  - [Set DeepSleep with timeout](#set-deepsleep-with-timeout)
+  - [Test with wake-up Source](#test-with-wake-up-source)
+
+## Acronyms, Terms and Abbreviations
+
+- `HAL`:    Hardware Abstraction Layer
+- `API`:   Application Programming Interface
+- `L2` - Level2 Testing
+- `L3` - Level3 Testing
+- `NA` : Not Applicable
+
+## Introduction
+
+This document provides an overview of the testing requirements for the Deep Sleep Manager module. It outlines the scope of testing, objectives, testing levels and approaches, specific test requirements, control plane emulator requirements and expected deliverables.
+
+Interface of the test is available here: [DeepSleep Manager HAL header](https://github.com/rdkcentral/rdk-halif-deepsleep_manager/blob/main/include/deepSleepMgr.h)
+
+## Module Description
+
+The Deep Sleep Manger `HAL` provides a set of `APIs` to initialize, set the deep sleep state and wake-up from the deep sleep state with a list of wake-up source.
+
+## Test Scenarios
+
+The Deep-sleep Manager layer facilitates the deep-sleep sleep and wake up procedures.
+
+|#|Test Functionality|Description|
+|-|------------------|-----------|
+|1|[Set deep sleep with timeout](#set-deep-sleep-with-timeout)|Set the deep sleep with timeout and verify the wake-up source |
+|2|[Test with Wake-up Source](#test-with-wake-up-source)|Configure the deep sleep mode with no timeout, ensuring that the Deep Sleep Manager facilitates wake-up sources, allowing the device to awaken from deep sleep.|
+
+
+### Set Deep Sleep with timeout
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+|1|[Set deep sleep with timeout](#set-deep-sleep-with-timeout)|Set the deep sleep with timeout and verify the wake-up source |PLAT_DS_SetDeepSleep|Y|`NA`|`NA`|
+
```

Comment:
We can add one more test case to verify if the wake up happened within the specified timeout value

Reply:
Added in an additional line. Not sure what it's not appearing here, but check line 44 in the file itself for the added test. Please review when you get a chance.

Reply:
Add the HAL APIs used for the test

Reply:
For the base test, it's _PLAT_DS_SetDeepSleep_, for the test to verify that's additiona test that's requested I don't think there is a function to call, since it'll automatically come out of deepsleep as per the time out, and we just need to record how long it was asleep for. (Likely by timestamping before and after the set deepsleep function call)

Reply:
for each test case we should mention the APIs. 
first test we are checking the wakeup soruce
second test we are checking the time taken to wakeup. 
both uses PLAT_DS_SetDeepSleep API. we should mention the PLAT_DS_SetDeepSleep in second test also

Reply:
I see what you mean. I've updated it as requested